contentcuration/contentcuration/views.py

```diff
@@ -160,13 +162,28 @@ def file_create(request):
 @csrf_exempt
 def thumbnail_upload(request):
     if request.method == 'POST':
-        filename = request.FILES.values()[0]._name
-        ext = os.path.splitext(filename)[1].split(".")[-1] # gets file extension without leading period
-        file_object = File(file_on_disk=request.FILES.values()[0], file_format_id=ext, preset_id=format_presets.CHANNEL_THUMBNAIL)
-        file_object.save()
+        fobj = request.FILES.values()[0]
+
+        # Check that hash is valid
+        checksum = hashlib.md5()
+        for chunk in iter(lambda: fobj.read(4096), b""):
+            checksum.update(chunk)
+        filename, ext = os.path.splitext(fobj._name)
+        hashed_filename = checksum.hexdigest()
+        full_filename = "{}{}".format(hashed_filename, ext)
+        fobj.seek(0)
+
+        # Get location of file
+        file_path = generate_file_on_disk_name(hashed_filename, full_filename)
+
+        # Write file if it doesn't already exist
+        with open(file_path, 'wb') as destf:
+            shutil.copyfileobj(fobj, destf)
```

Comment:
Seems like a lot of logic duplicated in here from the other file upload stuff. Can that be consolidated into a general purpose function?